From 4a9c9d05a2ef90e4dba14b1b5a33d9867ca814ea Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Wed, 28 Nov 2012 14:27:06 +0200
Subject: [PORT FROM MAIN] Telephony: Initialize Menu.defaultItem to -1

BZ: 63971

Valid value for defaultItem is from 0...n-1.
Initializing it to 0 will result in 1st item
in the menu selected even when the CAT hasn't
supplied any default item.

Initialize the defaultItem to -1 to avoid
the 1st item getting selected.

Change-Id: I58b4e802aec7e8e7e0cfcb1f7a2941ec86ffa7dc
Orig-Change-Id: I6cdf2be0876def4c4e7d074115e74a376731db6b
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/76946
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 .../com/android/internal/telephony/cat/Menu.java   |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/Menu.java b/src/java/com/android/internal/telephony/cat/Menu.java
index a3cbbdf..2673cc3 100644
--- a/src/java/com/android/internal/telephony/cat/Menu.java
+++ b/src/java/com/android/internal/telephony/cat/Menu.java
@@ -44,7 +44,7 @@ public class Menu implements Parcelable {
         items = new ArrayList<Item>();
         title = null;
         titleAttrs = null;
-        defaultItem = 0;
+        defaultItem = -1;
         softKeyPreferred = false;
         helpAvailable = false;
         titleIconSelfExplanatory = false;
-- 
1.7.4.1

